<a href="https://colab.research.google.com/github/raushan9jnv/Almabetter-Play-Store-App-Review-Analysis--Capstone-Project-1/blob/main/Almabetter_Play_Store_App_Review_Analysis_Capstone_Project_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

# **INTRODUCTION**

Play Store is a Application distribution platform and pre-installed on smartphones that ship with Google Mobile Services and it's free to use. It provides access to content on the Google Play Store, including apps, books, magazines, music, movies, and television programs.Their lots of designers and developers work on it to make play store services outstanding than other stores present in the market.

To take play store services in outstanding state, it would be great for developers and designer to analyze each and every aspect of current market like what people looking for, trend of a particular service, which type of desgin(UI) mostly liked by user and many more aspect. while considering these factors and work on it will affect the productivity of services. 

The objective of our exploratory data analysis project is to understand customer demands better and thus help developers and designers to improve their product engagement on the play store and make great impact on user.

# **OBJECTIVE**

In this data analysis project we have asked to analyze techanical aspect how about building a good android application and discover key factors responsible for app engagement and success. We also see trend and insights of play store.

We were given two datasets one is <b>Play Store Data</b> and other is <b>User Review</b>. before watching over data provided, let's first see what actually is Exploratory Data Analysis (EDA).

# **What is Exploratory Data Analysis?**

Exploratory Data Analysis (EDA) is an approach to analyzing datasets to summarize their main characteristics, often with visual methods. EDA is used for seeing what the data can tell us before the modeling task. It is not easy to look at a column of numbers or a whole spreadsheet and determine important characteristics of the data. It may be tedious, boring, and/or overwhelming to derive insights by looking at plain numbers. Exploratory data analysis techniques have been devised as an aid in this situation.

Exploratory data analysis is generally cross-classified in two ways. First, each method is either non-graphical or graphical. And second, each method is either univariate or multivariate (usually just bivariate)

The following steps involved in thev EDA process:
1. Identification of variables and data types
2. Analyzing the basic metrics
3. Non-Graphical Univariate Analysis
4. Graphical Univariate Analysis
5. Bivariate Analysis
6. Variable transformations
7. Missing value treatment
8. Outlier treatment
9. Correlation Analysis
10. Dimensionality Reduction

# **PROBLEM STATEMENT**

# **DATASET CONTENT**

###There are more than a million application which are available in google playstore.This dataset contain data of 10,000 apps from google playstore.

###<b>The Play Store dataset contain following features:</b>


1.   <b>App</b>- Name of the apps.
2.   <b>Category</b>- Category under which the app falls.
3.   <b>Rating</b>- Applications rating in playstore.
4.   <b>Reviews</b>- Number of reviews given to apps.
5.   <b>Size</b>- Size of the apps.
6.   <b>Installs</b>- Number of installs of the apps.
7.   <b>Type</b>- If the app is free or paid.
8.   <b>Price</b>- Price of the apps.
9.   <b>Content Rating</b>- Appropriate target audience of the apps.
10.  <b>Genres</b>- Genres under which the app fall.
11.  <b>Last updated</b>- Date when the app last updated.
12.  <b>Current Ver</b>- Current version of the apps.
13.  <b>Android Ver</b>- Minimum android version required for the apps to run.





###<b>The User Review dataset contain following features:</b>


1.   <b>App</b>- Name of the apps.

# **Time to Explore and Dive into Given Dataset**

In [27]:
# Let us begin with importing libraries

import pandas as pd # data manipulation
import numpy as np  # data manipulation 
import matplotlib.pyplot as plt # visualization tool
import seaborn as sns # visualization tool

from datetime import datetime
from datetime import date

%matplotlib inline

In [28]:
# Mounting Google Drive 

from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


#### I will start by importing the csv file into pandas dataframe called df_apps.  

In [29]:
file_path = "/content/drive/MyDrive/Almabetter/Capstone project/Capstone Project AlmaBetter EDA/"

In [30]:
# importing Play store and user review data

df_apps = pd.read_csv(file_path + "Play Store Data.csv")
df_reviews = pd.read_csv(file_path + "User Reviews.csv")

# Explore the Two DataFrames

In [31]:
# print the shape of Two Dataframe
print(f"The shape of Play Store dataframe is {df_apps.shape}")
print(f"The shape of User Review dataframe is {df_reviews.shape}")

The shape of Play Store dataframe is (10841, 13)
The shape of User Review dataframe is (64295, 5)


In [32]:
# print the columns of Two dataframes

print(f"The columns of Play Store Dataframe are \n \n {df_apps.columns}\n")
print(f"The columns of Play Store Dataframe are \n \n {df_reviews.columns}")


The columns of Play Store Dataframe are 
 
 Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

The columns of Play Store Dataframe are 
 
 Index(['App', 'Translated_Review', 'Sentiment', 'Sentiment_Polarity',
       'Sentiment_Subjectivity'],
      dtype='object')


In [33]:
# Filling the space between the column names with "_" for ease of use in the play_store.

df_apps.columns = df_apps.columns.str.replace(" ","_")
df_apps.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Current_Ver',
       'Android_Ver'],
      dtype='object')

# Lets Explore The Play Store Dataset

In [34]:
# print the 5 head and 5 row of Play Store dataframe

concat_Playstore = pd.concat([df_apps.head(5),df_apps.tail(5)])
concat_Playstore

App  ...         Android_Ver
0         Photo Editor & Candy Camera & Grid & ScrapBook  ...        4.0.3 and up
1                                    Coloring book moana  ...        4.0.3 and up
2      U Launcher Lite – FREE Live Cool Themes, Hide ...  ...        4.0.3 and up
3                                  Sketch - Draw & Paint  ...          4.2 and up
4                  Pixel Draw - Number Art Coloring Book  ...          4.4 and up
10836                                   Sya9a Maroc - FR  ...          4.1 and up
10837                   Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                             Parkinson Exercices FR  ...          2.2 and up
10839                      The SCP Foundation DB fr nn5n  ...  Varies with device
10840      iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10 rows x 13 columns]

In [35]:
# print rows and column of play store data

print(f"There are total {df_apps.shape[0]} rows and {df_apps.shape[1]} columns in playstore dataframe.")

There are total 10841 rows and 13 columns in playstore dataframe.


In [36]:
df_apps.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content_Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last_Updated    10841 non-null  object 
 11  Current_Ver     10833 non-null  object 
 12  Android_Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [37]:
df_apps.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Current_Ver',
       'Android_Ver'],
      dtype='object')

**Findings**

play_store dataframe has 10841 rows and 13 columns. The 13 columns are identified as below:


1.   <b>App</b>- Name of the apps.
2.   <b>Category</b>- Category under which the app falls.
3.   <b>Rating</b>- Applications rating in playstore.
4.   <b>Reviews</b>- Number of reviews given to apps.
5.   <b>Size</b>- Size of the apps.
6.   <b>Installs</b>- Number of installs of the apps.
7.   <b>Type</b>- If the app is free or paid.
8.   <b>Price</b>- Price of the apps.
9.   <b>Content Rating</b>- Appropriate target audience of the apps.
10.  <b>Genres</b>- Genres under which the app fall.
11.  <b>Last updated</b>- Date when the app last updated.
12.  <b>Current Ver</b>- Current version of the apps.
13.  <b>Android Ver</b>- Minimum android version required for the apps to run.




In [38]:
 df_apps.describe(include='all')

App Category  ...         Current_Ver Android_Ver
count    10841    10841  ...               10833       10838
unique    9660       34  ...                2832          33
top     ROBLOX   FAMILY  ...  Varies with device  4.1 and up
freq         9     1972  ...                1459        2451
mean       NaN      NaN  ...                 NaN         NaN
std        NaN      NaN  ...                 NaN         NaN
min        NaN      NaN  ...                 NaN         NaN
25%        NaN      NaN  ...                 NaN         NaN
50%        NaN      NaN  ...                 NaN         NaN
75%        NaN      NaN  ...                 NaN         NaN
max        NaN      NaN  ...                 NaN         NaN

[11 rows x 13 columns]

In [39]:
def PlaystoreInfo():
  PSinfo_df = pd.DataFrame(index=df_apps.columns)
  PSinfo_df['Datatypes'] =  df_apps.dtypes
  PSinfo_df['Count of non-null values'] = df_apps.count()
  PSinfo_df['NaN values'] = df_apps.isnull().sum()
  PSinfo_df['% NaN Values'] = (PSinfo_df['NaN values']/len(df_apps)).round(4)*100 # or df_apps.isnull().mean()
  PSinfo_df['Unique_count'] = df_apps.nunique()
  return PSinfo_df
PlaystoreInfo()

Datatypes  Count of non-null values  ...  % NaN Values  Unique_count
App               object                     10841  ...          0.00          9660
Category          object                     10841  ...          0.00            34
Rating           float64                      9367  ...         13.60            40
Reviews           object                     10841  ...          0.00          6002
Size              object                     10841  ...          0.00           462
Installs          object                     10841  ...          0.00            22
Type              object                     10840  ...          0.01             3
Price             object                     10841  ...          0.00            93
Content_Rating    object                     10840  ...          0.01             6
Genres            object                     10841  ...          0.00           120
Last_Updated      object                     10841  ...          0.00          1378
Current_Ver       object                     10833  ...          0.07          2832
Android_Ver       object                     10838  ...          0.03            33

[13 rows x 5 columns]

**Findings**

* From the above, we understand that **except for Rating columns**, we are having a good dataset.

* The number of null values are:
  1. **Rating** has 1474 null values which contributes **13.60%** of the data.
  2. **Type** has 1 null value which contributes **0.01%** of the data.
  3. **Content_Rating** has 1 null value which contributes **0.01%** of the data.
  4. **Current_Ver** has 8 null values which contributes **0.07%** of the data.
  5. **Android_Ver** has 3 null values which contributes **0.03%** of the data.

* We know that there is only **one feature with numeric type** i.e. Rating.
* Unique counts of **Type** and **Content ratings** are very small i.e. **3** and **6** respectively



# Lets Explore The User Review Dataset

In [40]:
# print the 5 head and 5 row of Play Store dataframe

concat_UserReview = pd.concat([df_reviews.head(5),df_reviews.tail(5)])
concat_UserReview

App  ... Sentiment_Subjectivity
0            10 Best Foods for You  ...               0.533333
1            10 Best Foods for You  ...               0.288462
2            10 Best Foods for You  ...                    NaN
3            10 Best Foods for You  ...               0.875000
4            10 Best Foods for You  ...               0.300000
64290  Houzz Interior Design Ideas  ...                    NaN
64291  Houzz Interior Design Ideas  ...                    NaN
64292  Houzz Interior Design Ideas  ...                    NaN
64293  Houzz Interior Design Ideas  ...                    NaN
64294  Houzz Interior Design Ideas  ...                    NaN

[10 rows x 5 columns]

In [41]:
# print rows and column of play store data

print(f"There are total {df_reviews.shape[0]} rows and {df_reviews.shape[1]} columns in playstore dataframe.")

There are total 64295 rows and 5 columns in playstore dataframe.


In [42]:
df_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


In [43]:
df_reviews.columns

Index(['App', 'Translated_Review', 'Sentiment', 'Sentiment_Polarity',
       'Sentiment_Subjectivity'],
      dtype='object')

**Findings**

user_reviews dataframe has 64295 rows and 5 columns. The 5 columns are identified as follows:
1. **App** - The name of the application.
2. **Translated_Review** - what the users feedback is about the application.
3. **Sentiment** - tells us about a view or opinion of the user w.r.t. the application.
4. **Sentiment_Polarity** - Sentiment polarity for an element defines the orientation of the expressed sentiment, i.e., it determines if the text expresses the positive, negative or neutral sentiment of the user about the application.
5. **Sentiment_Subjectivity** - It refers to the text that contains text which is usually expressed by a human having typical moods, emotions, and feelings. Mostly it is a public opinion and not a factual information. 

In [44]:
 df_reviews.describe(include='all')

App  ... Sentiment_Subjectivity
count                 64295  ...           37432.000000
unique                 1074  ...                    NaN
top     Angry Birds Classic  ...                    NaN
freq                    320  ...                    NaN
mean                    NaN  ...               0.492704
std                     NaN  ...               0.259949
min                     NaN  ...               0.000000
25%                     NaN  ...               0.357143
50%                     NaN  ...               0.514286
75%                     NaN  ...               0.650000
max                     NaN  ...               1.000000

[11 rows x 5 columns]

In [45]:
def UserReviewInfo():
  URinfo_df = pd.DataFrame(index=df_reviews.columns)
  URinfo_df['Datatypes'] =  df_reviews.dtypes
  URinfo_df['Count of non-null values'] = df_reviews.count()
  URinfo_df['NaN values'] = df_reviews.isnull().sum()
  URinfo_df['% NaN Values'] = (URinfo_df['NaN values']/len(df_reviews)).round(4)*100 # or df_REVIEWS.isnull().mean()
  URinfo_df['Unique_count'] = df_reviews.nunique()
  return URinfo_df
UserReviewInfo()

Datatypes  ...  Unique_count
App                       object  ...          1074
Translated_Review         object  ...         27994
Sentiment                 object  ...             3
Sentiment_Polarity       float64  ...          6195
Sentiment_Subjectivity   float64  ...          4530

[5 rows x 5 columns]

**Findings**

* From the above, we understand that **only for App column**, we are having a good dataset.
* There so many of null values in dataset, we need to take care of that part.

* The number of null values are:
  1. **Translated_Review** has 26868 null values which contributes **41.79%** of the data.
  2. **Sentiment** has 26863 null values which contributes **41.78%** of the data.
  3. **Sentiment_Polarity** has 26863 null values which contributes **41.78%** of the data.
  4. **Sentiment_Subjectivity** has 26863 null values which contributes **41.78%** of the data.

* We know that there is **two feature with numeric type** i.e. **Sentiment_Polarity** and **Sentiment_Subjectivity**.
* Unique count of **Sentiment**  are very small i.e. **3** .
* Most of the **Translated_review** and **Sentiment** are **Good** and **Positive**.



# **Data Cleaning - Univariate & Bivariate Analysis**

## Handling Missing Values

### Missing Values in user_review DF

In [46]:
UserReviewInfo()

Datatypes  ...  Unique_count
App                       object  ...          1074
Translated_Review         object  ...         27994
Sentiment                 object  ...             3
Sentiment_Polarity       float64  ...          6195
Sentiment_Subjectivity   float64  ...          4530

[5 rows x 5 columns]

In [47]:
# lets create new user review df, as we going to drop all null values
# We have dropped the rows having all null values as it makes no sense in keeping them
new_df_reviews = df_reviews.dropna(subset=["Translated_Review","Sentiment","Sentiment_Polarity","Sentiment_Subjectivity"],how="all")

In [48]:
new_df_reviews.shape

(37432, 5)

In [49]:
new_df_reviews.isnull().sum()

App                       0
Translated_Review         5
Sentiment                 0
Sentiment_Polarity        0
Sentiment_Subjectivity    0
dtype: int64

In [51]:
new_df_reviews[new_df_reviews.Translated_Review.isnull()]

App  ... Sentiment_Subjectivity
268                                         11st  ...                    0.0
15048        Birds Sounds Ringtones & Wallpapers  ...                    0.0
22092             Calorie Counter - MyFitnessPal  ...                    0.0
31623                                  DC Comics  ...                    0.0
52500  Garden Photo Frames - Garden Photo Editor  ...                    0.0

[5 rows x 5 columns]

In [52]:
new_df_reviews = new_df_reviews.dropna(subset= ["Translated_Review"],how = "all")
new_df_reviews.shape

(37427, 5)

In [54]:
def New_UserReviewInfo():
  URinfo_df = pd.DataFrame(index=new_df_reviews.columns)
  URinfo_df['Datatypes'] =  new_df_reviews.dtypes
  URinfo_df['Count of non-null values'] = new_df_reviews.count()
  URinfo_df['NaN values'] = new_df_reviews.isnull().sum()
  URinfo_df['% NaN Values'] = (URinfo_df['NaN values']/len(new_df_reviews)).round(4)*100 # or df_REVIEWS.isnull().mean()
  URinfo_df['Unique_count'] = new_df_reviews.nunique()
  return URinfo_df
New_UserReviewInfo()

Datatypes  ...  Unique_count
App                       object  ...           865
Translated_Review         object  ...         27994
Sentiment                 object  ...             3
Sentiment_Polarity       float64  ...          6195
Sentiment_Subjectivity   float64  ...          4530

[5 rows x 5 columns]

**CONCLUSION:** From the above output we can infer that we have removed all the NaN values from the user_review data frame.

### Missing Values in play_store DF